### DISCUSSION TITLE
Text Input Default Value from Database

### DISCUSSION THREAD

**Discussion by viamaze** [filamentphp/filament#12539] [Help]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

Hello,

I'm new to filament/laravel. I want to set a default value to textinput and this value should come from the database. Is there anyway I can do that?

Say for example, I want to retrieve a record that is only "active" in the database and I want that to be the default value of the text input.

Thanks for the help

**Comment by leandrocfe** [comment]
```php
->formatStateUsing(fn(?YourModel $record) => $record?->field ?? 'default')
```

**Reply by tahinuralam** [reply]
I have the same issues

**Reply by leandrocfe** [reply]
What issue?

**Reply by Raza9798** [reply]
use `->placeholder('Standard')` to render default values

**Comment by hairulloh-sukur** [comment]
You can use an approach like this to retrieve a value from the database:
`
->default(fn() => Company::find(1)->vat_percentage)
`
